src/arch/mod.rs

```diff
@@ -149,9 +160,11 @@ impl Type {
             Error => Err("cannot take `alignof` <type error>"),
         }
     }
+    /// Get the pointer type
     pub fn ptr_type() -> IrType {
         IrType::int(CHAR_BIT * PTR_SIZE).expect("pointer size should be valid")
     }
+    /// Convert `self` into the type for the IR.
```

Comment:
```suggestion
    /// Return an IR type which can represent this C type
```

'Convert' suggests that it consumes `self`.